src/pyedb/dotnet/edb_core/edb_data/components_data.py

```diff
@@ -297,6 +302,82 @@ def solder_ball_height(self):
             return self.component_property.GetSolderBallProperty().GetHeight()
         return None
 
+    @solder_ball_height.setter
+    def solder_ball_height(self, value):
+        if "GetSolderBallProperty" in dir(self.component_property):
+            if value:
+                sball_height = round(self._edb.utility.Value(value).ToDouble(), 9)
+                cmp_property = self.component_property
+                solder_ball_prop = cmp_property.GetSolderBallProperty().Clone()
+                solder_ball_prop.SetHeight(self._get_edb_value(sball_height))
+                cmp_property.SetSolderBallProperty(solder_ball_prop)
+                self.component_property = cmp_property
+
+    @property
+    def solder_ball_shape(self):
+        """Solder ball shape."""
+        if "GetSolderBallProperty" in dir(self.component_property):
+            shape = self.component_property.GetSolderBallProperty().GetShape()
+            if shape.value__ == 0:
+                return "None"
+            elif shape.value__ == 1:
+                return "Cylinder"
+            elif shape.value__ == 2:
+                return "Spheroid"
+
+    @solder_ball_shape.setter
+    def solder_ball_shape(self, value):
+        shape = None
+        if isinstance(value, str):
+            if value.lower() == "cylinder":
+                shape = self._edb.definition.SolderballShape.Cylinder
+            elif value.lower() == "none":
+                shape = self._edb.definition.SolderballShape.NoSolderball
+            elif value.lower() == "spheroid":
+                shape = self._edb.definition.SolderballShape.Spheroid
+        if isinstance(value, int):
+            if value == 0:
+                shape = self._edb.definition.SolderballShape.NoSolderball
+            if value == 1:
+                shape = self._edb.definition.SolderballShape.Cylinder
+            if value == 2:
+                shape = self._edb.definition.SolderballShape.Spheroid
+        if shape:
+            cmp_property = self.component_property
+            solder_ball_prop = cmp_property.GetSolderBallProperty().Clone()
+            solder_ball_prop.SetShape(shape)
+            cmp_property.SetSolderBallProperty(solder_ball_prop)
+            self.component_property = cmp_property
+
+    @property
+    def solder_ball_diameter(self):
+        """Solder ball diameter"""
+        if "GetSolderBallProperty" in dir(self.component_property):
+            result = self.component_property.GetSolderBallProperty().GetDiameter()
+            if result[0]:
+                return result[1], result[2]
```

Comment:
This is hardly readable. I assume that result[0] is a boolean but what are result[1] and [2] ? Are they "end" and "mid" diameters ?

Reply:
Yes this is the output from internal EDB api

Reply:
result[1] and [2] are 2 diameters (main and mid one) 
![image](https://github.com/ansys/pyedb/assets/74993647/ad78986d-977f-4ac2-b8c7-94d2ff6354d3)
Using overload also